client-ts/Microsoft.AspNetCore.SignalR.Client.TS.Tests/HubConnection.spec.ts

```diff
@@ -2,150 +2,290 @@ import { IConnection } from "../Microsoft.AspNetCore.SignalR.Client.TS/IConnecti
-        // Typically this would be called by the transport
-        connection.onClosed(new Error("Connection lost"));
-
-        try {
-            await invokePromise;
-            fail();
-        }
-        catch (e) {
-            expect(e.message).toBe("Connection lost");
-        }
-        done();
-    });
+            let hubConnection = new HubConnection(connection);
+            var invokePromise = hubConnection.invoke("testMethod", "foo", 42)
+                .catch((_) => { }); // Suppress exception and unhandled promise rejection warning.
 
-    it("sends invocations as nonblocking", async done => {
-        let dataSent: string;
-        let connection: IConnection = {
-            start(transportType: TransportType): Promise<void> {
-                return Promise.resolve();
-            },
+            // Verify the message is sent
+            expect(connection.sentData.length).toBe(1);
+            expect(JSON.parse(connection.sentData[0])).toEqual({
+                type: 1,
+                invocationId: connection.lastInvocationId,
+                target: "testMethod",
+                nonblocking: false,
+                arguments: [
+                    "foo",
+                    42
+                ]
+            });
 
-            send(data: any): Promise<void> {
-                dataSent = data;
-                return Promise.resolve();
-            },
+            // Close the connection
+            hubConnection.stop();
+        });
 
-            stop(): void {
-                if (this.onClosed) {
-                    this.onClosed();
-                }
-            },
+        it("rejects the promise when an error is received", async () => {
+            let connection = new TestConnection();
 
-            onDataReceived: null,
-            onClosed: null
-        };
+            let hubConnection = new HubConnection(connection);
+            var invokePromise = hubConnection.invoke("testMethod", "foo", 42);
 
-        let hubConnection = new HubConnection(connection);
-        let invokePromise = hubConnection.invoke("testMethod");
+            connection.receive({ type: 3, invocationId: connection.lastInvocationId, error: "foo" });
 
-        expect(JSON.parse(dataSent).nonblocking).toBe(false);
+            let ex = await captureException(async () => invokePromise);
+            expect(ex.message).toBe("foo");
+        });
 
-        // will clean pending promises
-        connection.onClosed();
+        it("resolves the promise when a result is received", async () => {
+            let connection = new TestConnection();
 
-        try {
-            await invokePromise;
-            fail(); // exception is expected because the call has not completed
-        }
-        catch (e) {
-        }
-        done();
+            let hubConnection = new HubConnection(connection);
+            var invokePromise = hubConnection.invoke("testMethod", "foo", 42);
```

Comment:
Either remove these parameters or rename "foo" to something else in tests where the expected result or error is also "foo".